### PR TITLE
Implement power service

### PR DESCRIPTION
**Pull request by sfc-gh-ssudakovich**
Implement power service

It is useful to be able to reboot a system when necessary. 
To implement that on a linux system, we rely on /sbin/shutdown
command. Other system are not implemented.

### CODE REVIEW

File: services/power/client/client.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
+/* Copyright (c) 2019 Snowflake Inc. All rights reserved.
```
Comment (sfc-gh-elinardi) [inline services/power/client/client.go:1]: Change year to 2023?

File: services/power/client/client.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	c := client.SetupSubpackage(subPackage, f)
+	c.Register(&rebootCmd{}, "")
+	return c
+}
+
+func (*powerCmd) Name() string { return subPackage }
+func (p *powerCmd) Synopsis() string {
+	return client.GenerateSynopsis(p.GetSubpackage(flag.NewFlagSet("", flag.ContinueOnError)), 2)
+}
+func (*powerCmd) Usage() string { return "" }
```
Comment (sfc-gh-elinardi) [inline services/power/client/client.go:51]: `return client.GenerateUsage(subPackage, p.Synopsis())` might be a better than empty

File: cmd/sanssh/main.go [hunk 3]
```diff
@@ -46,6 +46,7 @@ import (
 	_ "github.com/Snowflake-Labs/sansshell/services/healthcheck/client"
 	_ "github.com/Snowflake-Labs/sansshell/services/localfile/client"
 	_ "github.com/Snowflake-Labs/sansshell/services/packages/client"
+	_ "github.com/Snowflake-Labs/sansshell/services/power/client"
```
Comment (sfc-gh-elinardi) [inline cmd/sanssh/main.go:49]: it's not too important, but good idea to also add this to cmd/sansshell-server/main.go

File: services/power/client/client.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	return `reboot:
+  Reboot a host at a predefined time.
+ `
+}
+func (a *rebootCmd) Usage() string {
+	return client.GenerateUsage(subPackage, a.Synopsis())
+}
+
+func (a *rebootCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&a.when, "when", "now", "When to reboot a host. Valid values are 'now', 'hh:mm', '+5m'.")
```
Comment (sfc-gh-elinardi) [inline services/power/client/client.go:76]: `+5m` won't work, so perhaps change this to "`+m` where m is the number of minutes "

File: services/power/client/client.go (hunk 1 above)
Comment (sfc-gh-ssudakovich) [reply services/power/client/client.go]: Thanks, I did forget what year we live in :-)

File: services/power/client/client.go (hunk 4 above)
Comment (sfc-gh-ssudakovich) [reply services/power/client/client.go]: Thanks. I should not advertise what is obviously wrong

File: cmd/sanssh/main.go (hunk 3 above)
Comment (sfc-gh-ssudakovich) [reply cmd/sanssh/main.go:49]: ack

**Review by sfc-gh-elinardi — APPROVED** [review]

**Pull request merged**